jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/RunConfigurationCredentialsExtension.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.execution.CommonJavaRunConfigurationParameters
+import com.intellij.execution.RunConfigurationExtension
+import com.intellij.execution.configurations.JavaParameters
+import com.intellij.execution.configurations.RunConfigurationBase
+import com.intellij.execution.configurations.RunnerSettings
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.util.Key
+import com.intellij.ui.layout.panel
+import com.intellij.util.xmlb.XmlSerializer
+import org.jdom.Element
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.resources.message
+import javax.swing.JCheckBox
+import javax.swing.JComponent
+
+class RunConfigurationCredentialsExtension : RunConfigurationExtension() {
+    override fun isApplicableFor(configuration: RunConfigurationBase<*>): Boolean = configuration is CommonJavaRunConfigurationParameters
+
+    override fun <T : RunConfigurationBase<*>?> updateJavaParameters(configuration: T, params: JavaParameters, runnerSettings: RunnerSettings?) {
```

Comment:
using `com.intellij.execution.RunConfigurationExtension#patchCommandLine` would probably be safer since it is more generic

Reply:
you *have* to implement the `updateJavaParameters` but yeah for the python implementing that would proly be better.